src/getThreadList.js

```diff
@@ -1,73 +1,211 @@
-        callback = type;
-        type = "inbox"; //default to inbox
-      } else if (utils.getType(type) !== "String") {
-        throw {
-          error:
-            "Please pass a String as a third argument. Your options are: inbox, pending, and archived"
+      case "ReducedMessagingActor":
+        return {
+          accountType: p["__typename"],
+          userID: utils.formatID(p.id.toString()),
+          name: p.name,
+          url: createProfileUrl(p.url, p.username, p.id), // in this case p.url is null all the time
+          profilePicture: p.big_image_src.uri, // in this case it is default facebook photo, we could determine gender using it
+          username: (p.username||null), // maybe we could use it to generate profile URL?
+          isMessageBlockedByViewer: p.is_message_blocked_by_viewer, // true/false
         };
-      } else {
-        throw {
-          error: "getThreadList: need callback"
+      case "UnavailableMessagingActor":
+        return {
+          accountType: p["__typename"],
+          userID: utils.formatID(p.id.toString()),
+          name: p.name, // "Facebook User" in user's language
+          url: createProfileUrl(p.url, p.username, p.id), // in this case p.url is null all the time
+          profilePicture: p.big_image_src.uri, // default male facebook photo
+          username: (p.username||null), // maybe we could use it to generate profile URL?
+          isMessageBlockedByViewer: p.is_message_blocked_by_viewer, // true/false
+        };
+      default:
+        log.warn("getThreadList", "Found participant with unsupported typename. Please open an issue at https://github.com/Schmavery/facebook-chat-api/issues\n" + JSON.stringify(p, null, 2));
+        return {
+          accountType: p["__typename"],
+          userID: utils.formatID(p.id.toString()),
+          name: p.name || `[unknown ${p["__typename"]}]`, // probably it will always be something... but fallback to [unknown], just in case
         };
-      }
     }
+  });
+}
 
-    if (type === "archived") {
-      type = "action:archived";
-    } else if (type !== "inbox" && type !== "pending" && type !== "other") {
-      throw {
-        error:
-          "type can only be one of the following: inbox, pending, archived, other"
-      };
-    }
+// "FF8C0077" -> "8C0077"
+function formatColor(color) {
+  if (color && color.match(/^(?:[0-9a-fA-F]{8})$/g)) {
+    return color.slice(2);
+  }
+  return color;
+}
 
-    if (end <= start) end = start + 20;
+function getThreadName(t) {
+  if (t.name || t.thread_key.thread_fbid) return t.name;
 
-    var form = {
-      client: "mercury"
-    };
+  for (let p of t.all_participants.nodes) {
+    if (p.messaging_actor.id === t.thread_key.other_user_id) return p.messaging_actor.name;
+  }
+}
 
-    form[type + "[offset]"] = start;
-    form[type + "[limit]"] = end - start;
+function mapNicknames(customizationInfo) {
+  return customizationInfo?customizationInfo.participant_customizations.map(u => {
```

Comment:
can you do `customizationInfo && customizationInfo.participant_customizations ? ...`